loilo/livy
# Rearchitect project in preparation for v2

# Architectural Preparation for v2

There have been major changes to the project structure:

- require Node.js v18+
- support ESM only (CommonJS support dropped)
- ~migrate from yarn workspaces to npm workspaces~
  For some reason, GitHub Actions _really_ does not want to run `npm ci` (where it works completely fine locally). Therefore I'm returning to using yarn for now.
- migrate from Jest to Vitest
- update all dependencies (probably breaking due to changed behaviors of
  dependencies and increased Node.js version requirements)
- update all dev dependencies, including
  - TypeScript to latest version
  - Lerna to v7 with Nx caching enabled
  - Latest versions of used ESLint plugins/presets
- Restructure npm scripts (move testing, linting and building into
  individual package.json files) to make Nx caching possible

To Do:
- Research how to keep Nx cache (and possibly also installed `node_modules`) between individual GitHub Actions runs to speed up CI
- Figure out migration steps for update from v1 (see "Dependency Updates" section)

## Depdency Updates

These are the dependency updates in individual components. Listed are only updates that might affect the public API/the side effects of the package.

These updates affect `@livy/contracts` (and therefore all `@livy/*` packages):

Package | Old version | New version
-|-|-
`luxon` | `^1.25.0` | `^3.2.0`
`typescript` | `^4.0.3` | `^5.1.0`
`@types/node` | `^14.11.2` | `^18.11.18`

### `@livy/console-formatter`

Package | Old version | New version
-|-|-
`emphasize` | `^4.0.0` | `^6.0.0`
`js-yaml` | `^3.13.1` | `^4.0.0`

### `@livy/html-oneline-formatter`

Package | Old version | New version
-|-|-
`highlight.js` | `^10.2.0` | `^11.7.0`
`lowlight` | `^1.15.0` | `^2.8.0`

### `@livy/html-pretty-formatter`

Package | Old version | New version
-|-|-
`js-yaml` | `^3.13.1` | `^4.0.0`
`lowlight` | `^1.15.0` | `^2.8.0`

### `@livy/http-handler`

Package | Old version | New version
-|-|-
`got` | `^11.7.0` | `^12.5.3`
